src/applications/personalization/dashboard/tests/components/CTALink.unit.spec.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+
+import CTALink from '../../components/CTALink';
+
+describe('<CTALink />', () => {
+  it('should render', () => {
+    const linkText = 'Testing link';
+    const tree = render(
+      <CTALink testId="12345" href="https://example.com" text={linkText} />,
+    );
+
+    expect(tree.getByTestId('12345')).to.not.be.null;
+    expect(tree.getByText(linkText)).to.not.be.null;
+    expect(tree.container.querySelector('i.fa-chevron-right')).to.be.null;
+    tree.unmount();
+  });
+
+  it('should render the correct href', () => {
+    const wrapper = shallow(
+      <CTALink testId="12345" href="https://example.com" text="Testing link" />,
+    );
+
+    expect(wrapper.prop('href')).to.eq('https://example.com');
+    wrapper.unmount();
+  });
+
+  it('should show the arrow', () => {
+    const tree = render(
+      <CTALink
+        testId="12345"
+        href="https://example.com"
+        text="Testing link"
+        showArrow
+      />,
+    );
+
+    expect(tree.container.querySelector('i.fa-chevron-right')).to.not.be.null;
+    tree.unmount();
+  });
+
+  it('should open in a new tab', () => {
+    const wrapper = shallow(
+      <CTALink
+        testId="12345"
+        href="https://example.com"
+        text="Testing link"
+        newTab
+      />,
+    );
+
+    expect(wrapper.prop('target')).to.eq('_blank');
+    expect(wrapper.prop('rel')).to.eq('noreferrer noopener');
+    wrapper.unmount();
+  });
+
+  it('should render the aria-label', () => {
+    const wrapper = shallow(
+      <CTALink
+        testId="12345"
+        href="https://example.com"
+        text="Testing link"
+        ariaLabel="test label"
+      />,
+    );
+
+    expect(wrapper.prop('aria-label')).to.eq('test label');
+    wrapper.unmount();
+  });
```

Comment:
I think we are trying to stay away from using Enzyme in general, and instead using the React Testing Library render() instead where possible.

A potential way to rewrite this test might be:

``` js
  it('should render the aria-label', () => {
    const view = render(
      <CTALink
        testId="12345"
        href="https://example.com"
        text="Testing link"
        ariaLabel="test label"
      />,
    );

    expect(view.getByLabelText('test label')).to.exist;
  });
```

Also note, you don't have to call view.unmount() when using the React Testing Library as that is used more to test behavior specific to when a component unmounts. With Enzyme you used to have to cleanup after yourself with the unmount, but RTL takes care of that for you.

Reply:
@dellerbie you are 100% correct it is tough/near impossible to test those types of things with React Testing LIbrary. I guess their philosophy is to test the components in a way that the user is using them, but that definitely results in a lower test coverage because you cant cover absolutely prop/function. 

I guess it is a balance between testing implementation details and testing the desired functionality of a component from the user's perspective. I think it's fine to use Enzyme here and there if you are looking to test specific behaviors though.

Have you watched this video on unit testing for VA? https://www.youtube.com/watch?v=8Y0cuMUoWAw it might be useful 🤷🏼 it is from the [unit testing guidelines section of the docs.](https://depo-platform-documentation.scrollhelp.site/developer-docs/unit-tests)

Reply:
I'm not sure this is a 'better' way to assert on a specific attribute, but I did discover a way to do this by looking at some other unit tests in some apps.

``` js
// top o file import of helper
import { $$ } from '~/platform/forms-system/src/js/utilities/ui';

// use the container that available from RTL and then query by attr and asert on length found
  it('should render the correct href', () => {
    const { container } = render(
      <CTALink testId="12345" href="https://example.com" text="Testing link" />,
    );

    expect($$('a[href="https://example.com"]', container).length).to.equal(1);
  });
```

Seems a bit of a backwards way to do this type of testing, so I'm fine with either RTL or Enzyme being used, but thought I would offer an alternative for future consideration.